io/include/detray/io/common/detail/file_handle.hpp

```diff
@@ -10,9 +10,9 @@
 // System include(s)
 #include <cassert>
 #include <cstdint>
+#include <filesystem>
```

Comment:
:confused: I'm surprised that this worked...

Using `<filesystem>` with GCC (but also with the other compilers) is a pain still. :frowning: One needs to link against `libstdc++fs.so` to use that header. At least I thought that was still needed.

But I don't see any CMake update in this PR that would've added a link against that library. :thinking:

In any project that already depends on Boost, the recommendation "from ATLAS" is to just use [boost::filesystem](https://www.boost.org/doc/libs/1_82_0/libs/filesystem/doc/index.htm) since using that is quite a bit "cleaner" on a build system level than using `std::filesystem`.

But since Detray does not use Boost yet, I wouldn't want to add a dependency just for this. :thinking:

I guess this is mainly a note that we should be prepared for this dependency to cause us problems down the line. :frowning:

Reply:
I was not aware. It seems it is fine with never versions of gcc and libstdc++. Apart from boost, what might be an alternative?

Reply:
I'm only aware of "other ATLAS code" having used those two options.

It's not impossible that the `std::filesystem` operations used in the code do not require `-lstdc++fs`. I don't know exactly which operations do and do not. :thinking:

As long as the CI is happy, I guess this is fine for now. I mainly wanted to make everyone aware that in other projects we've been trying to use `std::filesystem` so far.

Note that I complained about similar issues in other places as well. :stuck_out_tongue:
  - https://github.com/acts-project/traccc/pull/68
  - https://github.com/acts-project/traccc/pull/264
  - https://github.com/acts-project/acts/pull/423

(Just to point out a few...)